acrn: do not fail on non-ACRN systems

tttech-industrial-buchsbaum opened the pull request:
To be able to run libvirt on any system we must not fail if there is no
ARCN hypervisor available. Simply skip initialization without failure
in this case.

Signed-off-by: Helmut Buchsbaum <helmut.buchsbaum@opensource.tttech-industrial.com>

peterfang commented on src/acrn/acrn_driver.c:
```diff
@@ -2562,11 +2562,21 @@ acrnStateInitialize(bool privileged,
+        return 0;
+    }
+    close(fd);
+
```
Since acrnInitPlatform() is called after this, how about if we do the following?

- make acrnGetPlatform() return error codes (and perhaps -ENODEV in this case)
- make acrnInitPlatform() return error codes
- print a message and return 0 when acrnInitPlatform() indicates this is not an ACRN system

tttech-industrial-buchsbaum committed:
acrn_driver: use error codes in acrnGetPlatform

Return approproiate error codes in acrnGetPlatform to be able to handle
the errors differently on caller's side.

Signed-off-by: Helmut Buchsbaum <helmut.buchsbaum@opensource.tttech-industrial.com>

tttech-industrial-buchsbaum committed:
acrn_driver: use error codes in acrnInitPlatform

Return appropriate error codes in acrnInitPlatform to be able to handle
the errors differently on caller's side. In particular propagate
error code when possible.

Signed-off-by: Helmut Buchsbaum <helmut.buchsbaum@opensource.tttech-industrial.com>

tttech-industrial-buchsbaum committed:
acrn: do not fail on non-ACRN systems

To be able to run libvirt on any system we must not fail if there is no
ARCN hypervisor available. Simply skip initialization with proper cleanup
but without failure in this case.

Signed-off-by: Helmut Buchsbaum <helmut.buchsbaum@opensource.tttech-industrial.com>

tttech-industrial-buchsbaum commented:
I just updated the patches according to Peters suggestions. Please double-check if the used return codes are appropriate in all cases!

peterfang commented on src/acrn/acrn_driver.c:
```diff
@@ -2593,8 +2595,14 @@ acrnStateInitialize(bool privileged,
+    if (ret == -ENODEV) {
+        /* we are not running on an ACRN enabled system */
+        VIR_INFO("ACRN hypervisor not available, disabling driver");
+        return 0;
```
Can we goto cleanup here as well? virMutexInit() was already called previously so it'd be more graceful to destroy it. We still need to return 0 in this case.

tttech-industrial-buchsbaum commented:
PATCHv3: Adapted the patch to gracefully exit acrnStateInitialize().

peterfang reviewed (approved):
Looks good. Thanks!
